gcs/doc.md

```diff
@@ -0,0 +1,147 @@
+
+### Steps:
+#### Create a GCS bucket
+1. Login to your Google Cloud account and navigate to the GCS console
+2. Click on the "Create bucket" button and follow the prompts to create a new bucket
+_Note the name of your bucket as you will need it later_
+
+#### Set up Google Cloud credentials
+1. Create a new service account by navigating to the IAM & Admin console and selecting "Service accounts" from the left-hand menu
+2. Click on the "Create Service Account" button and follow the prompts to create a new service account
+3. Once you have created the service account, navigate to the "Keys" tab and click on the "Create Key" button
+4. Select "JSON" as the key type and download the JSON key file
+_Note the path to your JSON key file as you will need it later_
+
+#### Install Google Cloud SDK
+1. Download and install the Google Cloud SDK from the official website
+2. Follow the prompts to set up the SDK on your local machine
+3. Authenticate the SDK using the command `gcloud auth login`
+
+## Mounting a Bifrost Bridge to GCS
+1. Install Bifrost using: ```go get github.com/bifrost-cloud/bifrost```
+2. Initialize a new Bifrost client and mount a GCS bridge using the following code:
+```go
+package main
+
+import (
+	"fmt"
+
+	"github.com/opensaucerer/bifrost"
+)
+
+func main() {
+	bridge, err := bifrost.NewRainbowBridge(&bifrost.BridgeConfig{
+		DefaultBucket:   "bifrost",
+		DefaultTimeout:  10,
+		Provider:        bifrost.GoogleCloudStorage,
+		CredentialsFile: "/path/to/service/account/json", // this is not required if you are using google's default credentials
+		EnableDebug:     true,
+		PublicRead:      true,
+	})
+	if err != nil {
+		// bifrost comes with some error codes
+		if err.(bifrost.Error).Code() == bifrost.ErrInvalidProvider {
+			fmt.Println("Whoops, you didn't specify a valid provider!")
+			return
+		}
+		fmt.Println(err.(bifrost.Error).Code(), err)
+		return
+	}
+	defer bridge.Disconnect()
+	fmt.Printf("Connected to %s\n", bridge.Config().Provider)
+}
+```
+And that's it! You have now mounted a Bifrost bridge to your GCS account and can start uploading files using Bifrost.
+
+## Shipping a file to Google Cloud Storage via the rainbow bridge
+Now that you have mounted a Bifrost bridge to GCS, you can use Bifrost to upload files to GCS using the following code:
+```go
+// Upload a file
+uploadedFile, err := bridge.UploadFile(bifrost.File{
+	Path:     "../shared/image/aand.png",
+	Filename: "a_and_ampersand.png",
+	Options: map[string]interface{}{
+		bifrost.OptMetadata: map[string]string{
+			"originalname": "aand.png",
+		},
+	},
+})
+if err != nil {
+	fmt.Println(err)
+	return
+}
+fmt.Printf("Uploaded file: %s to %s\n", uploadedFile.Name, uploadedFile.Preview)
+```
+As you can see, uploading a file to GCS using Bifrost is as simple as calling the UploadFile method on the Bifrost client with the name of the bucket, the path to the file on your local machine, and the name of the file on GCS.

```

Comment:
Actually, you don't specify the bucket name in the call to `UploadFile` or any other function at that. One single Bifrost client interacts with one single bucket defined while instantiating the client.